controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java

```diff
@@ -148,8 +148,11 @@ public void createReaderGroup(ReaderGroupConfiguration request, StreamObserver<C
         RequestTag requestTag = requestTracker.initializeAndTrackRequestTag(requestIdGenerator.get(), "createReaderGroup",
                 scope, rgName);
         log.info(requestTag.getRequestId(), "createReaderGroup called for ReaderGroup {}/{}.", scope, rgName);
+        AuthHandler.Permissions requiredPermission = this.isRGStreamWritesWithReadPermEnabled ?
```

Comment:
Elsewhere in this class, we have used `StreamAuthParams` to abstract out these details. You might want to use an object of that class to delegate this auth-specific business logic to that abstraction. 

See an example here: https://github.com/pravega/pravega/blob/466d71a06cfb0b37f9163a52073b45f050c91b77/controller/src/main/java/io/pravega/controller/server/rpc/grpc/v1/ControllerServiceImpl.java#L305

Reply:
@anirudhkovuru later found out the `StreamAuthParams` won't work in this case, which only currently handles reader groups coming as internal streams.